src/api/services/ListingItemTemplateService.ts

```diff
@@ -419,6 +426,40 @@ export class ListingItemTemplateService {
         return messageSize;
     }
 
+    // sets an image as a "featured" image
+    public async setFeaturedImg(listingItemTemplate: resources.ListingItemTemplate, imageID: number): Promise<void> {
+        try {
+            const templateID = listingItemTemplate.id;
+            const itemImages = listingItemTemplate.ItemInformation.ItemImages;
+            if (!itemImages.find((img) => img.id === imageID)) {
+                throw new MessageException('Image ID doesnt exist on template');
+            }
+            if (itemImages) {
+                // loop through templates to check for previous featured images, sets to false
+                for (const item of itemImages) {
+                    if (item.featuredImg && item.id !== imageID) {
+                        const data = {
+                            id: item.id,
+                            featured_img: false
+                        };
+                        await this.itemImageRepo.update(templateID, data);
+                    }
+                    if (item.id === imageID && !item.featuredImg) {
+                        const data = {
+                            id: item.id,
+                            featured_img: true
+                        };
+                        await this.itemImageRepo.update(templateID, data);
+                    }
+                }
+                this.log.info('Successfully set featured image');
+            }
+        } catch (error) {
+            this.log.error(error);
+            throw new MessageException('Failed to set featured image');
+        }
+    }
+
```

Comment:
combining all the comments here, I'd change this to something like this:
    /**
     * sets the featured image for the template
     *
     * @param listingItemTemplate
     * @param imageId
     */
    public async setFeaturedImage(listingItemTemplate: resources.ListingItemTemplate, imageId: number): Promise<ItemImage> {
        const itemImages = listingItemTemplate.ItemInformation.ItemImages;

        if (!_.isEmpty(itemImages)) {
            const newFeaturedImage = itemImages.find((img) => img.id === imageId);
            if (!newFeaturedImage) {
                throw new MessageException('Image does not belong to the ListingItemTemplate.');
            }
            // unfeature the old one, if it exists
            const oldFeaturedImage = itemImages.find((img) => img.featured === true);
            if (oldFeaturedImage) {
                await this.itemImageService.updateFeatured(oldFeaturedImage.id, false);
            }
            // then feature the new one
            return await this.itemImageService.updateFeatured(newFeaturedImage.id, true);
        } else {
            throw new MessageException('ListingItemTemplate has no Images.');
        }
    }

this would go in ItemImageService:
    public async updateFeatured(id: number, featured: boolean): Promise<ItemImage> {
        return await this.itemImageRepo.update(id, {featured}); // update uses patch so just setting the feature should be enough
    }